### ISSUE TITLE
Racket 6.10: raco pkg install fails to connect to servers in TrueOS (FreeBSD -CURRENT)/LibreSSL

### ISSUE THREAD

**Issue by NorwegianRockCat** [racket/racket#1812]
Hello,
I'm getting an error when with raco pkg install with Racket 6.10 on TrueOS (unstable). I'm using the Racket package that is built from TrueOS ports tree.

This worked fine with Racket 6.9 (and continues to work fine on 6.9 with the same machine).

The error I get looks like this:

```
/usr/local/bin/raco pkg install pollen
Resolving "pollen" via https://pkgs.racket-lang.org
tcp-write: error writing
  system error: Socket is not connected; errno=57
  context...:
   /usr/local/share/racket/collects/openssl/mzssl.rkt:985:0: pump-output-once
   /usr/local/share/racket/collects/openssl/mzssl.rkt:1441:8: loop
   /usr/local/share/racket/collects/openssl/../racket/private/more-scheme.rkt:261:28
   /usr/local/share/racket/collects/openssl/../racket/contract/private/arrow-val-first.rkt:306:25
   /usr/local/share/racket/collects/net/http-client.rkt:262:0
   /usr/local/share/racket/collects/racket/contract/private/arrow-val-first.rkt:397:3
   /usr/local/share/racket/collects/net/url.rkt:187:0: http://getpost-impure-port
   /usr/local/share/racket/collects/net/url.rkt:294:2: redirection-loop
   /usr/local/share/racket/collects/racket/contract/private/arrow-val-first.rkt:397:3
   /usr/local/share/racket/collects/pkg/private/network.rkt:59:3
   /usr/local/share/racket/collects/pkg/private/catalog.rkt:218:0: read-from-server
   /usr/local/share/racket/collects/pkg/private/catalog.rkt:138:9: for-loop
   /usr/local/share/racket/collects/pkg/private/catalog.rkt:135:2: lookup-normally
   /usr/local/share/racket/collects/pkg/private/../../racket/private/more-scheme.rkt:261:28
   /usr/local/share/racket/collects/pkg/private/prefetch.rkt:128:2
   /usr/local/share/racket/collects/pkg/private/catalog.rkt:132:0: package-catalog-lookup9
```

My Racket skills fail here.

TrueOS is basically a snapshot of FreeBSD -current, but it also uses LibreSSL (version 2.5.5) instead of OpenSSL, I'm not sure if it that is pertinent, but since it looks like it's in the SSL code, it might be relevant.

Granted I feel bad about submitting a bug report for from "unstable" code in FreeBSD/TrueOS. If I had more machines around, I'd try and see if the same issue is happening in FreeBSD 11-Stable w/LibreSSL, but alas I don't. I can help a little bit with syncing and compiling if someone wants to try a patch. 

I sent the same report to the port maintainer at FreeBSD, but got no response. So, I thought I would try the source.

For the moment I have to still use Racket 6.9.

**Comment by NorwegianRockCat** [comment]
I tried this on another laptop over the weekend with the same version of the OS, and it all worked fine. The only difference between them is that the one that works is T420 and the one that doesn't work is the T560.

So, raco pkg works on one and not on another. I've copied the packages over to the non-working machine and can continue working, but I'm not sure how valid the bug report is at the moment. I'll leave it here and let you make a decision.
